src/test/java/javaslang/collection/euler/ProjectEulerTest.java

```diff
@@ -85,9 +84,37 @@ private static long fibonacci(int order) {
      * <p>See also <a href="https://projecteuler.net/problem=7">projecteuler.net problem 7</a>.</p>
      */
     @Test
-    @Ignore
-    public void shouldSolveProblem7() {
-        final int actual = Primes.asStream().drop(10_000).head();
-        assertThat(actual).isEqualTo(0); // TODO, see Primes#sieve(Stream)
+    public void test7() {
+        assertThat(primeNo(1)).isEqualTo(2);
+        assertThat(primeNo(2)).isEqualTo(3);
+        assertThat(primeNo(3)).isEqualTo(5);
+        assertThat(primeNo(4)).isEqualTo(7);
+        assertThat(primeNo(5)).isEqualTo(11);
+        assertThat(primeNo(6)).isEqualTo(13);
+        assertThat(primeNo(10_001)).isEqualTo(104_743);
+    }
+
+    private static int primeNo(int index) {
+        return Match
+                .when(1, (i) -> 2)
+                .orElse(() -> knownPrimes.drop(index - 1).head())
+                .apply(index);
+    }
+
+    private static final Stream<Integer> knownPrimes = Stream.gen(2, (p) -> nextPrime(p));
+
+    private static int nextPrime(int previousPrime) {
+        final Integer nextPrime = Match
```

Comment:
Btw (offtopic) - the Match API will change slightly with #227. The current `orElse` will be named `otherwise`. And there will be an _additional_, more monadic API that returns a value instead of a function which has to be applied.

```java
Match.of(value)
   .when(...)
   .when(...)
   .otherwise(...);
```